features/steps/validation_handling.py

```diff
@@ -163,20 +181,21 @@ def apply_then_operation(fn, inst, context, current_path, depth=0, **kwargs):
                     else validation_outcome.expected)
 
                 context.gherkin_outcomes.append(validation_outcome)
-
-            if not step_results:
-
-                validation_outcome = ValidationOutcome(
-                    outcome_code=ValidationOutcomeCode.PASSED,  # todo @gh "Rule passed" # deactivated until code table is added to django model
-                    observed=None,
-                    expected=None,
-                    feature=context.feature.name,
-                    feature_version=misc.define_feature_version(context),
-                    severity=OutcomeSeverity.PASSED,
-                    instance_id = safe_method_call(activation_inst, 'id', None),
-                    validation_task_id=context.validation_task_id
-                )
-                context.gherkin_outcomes.append(validation_outcome)
+            
+            # Currently, we should not inject passed outcomes for each individual instance to the databse
+            # if not step_results:
```

Comment:
But this is not for individual instances right, there is no loop? This was for the file-wide notification that for a certain feature at least one then step passed. So that in the UI/DB we can take the highest severity,

Reply:
No, I think we agreed that adding an entry for every outcome for each instance would make the database too heavy. Currently, we only use '`passed`' in Gherkin between given statements.
Therefore, the out-commented code can probably be deleted.

`context.instances = list(map(attrgetter('instance_id'), filter(lambda res: res.severity == OutcomeSeverity.PASSED, insts)))`

We could define a rule as _passed_ if it is activated, unless a `fail` outcome is found. In other words, if the highest severity is '`executed`,' we consider the file to have passed the rule.

It might be even clearer to use 'activated' between steps, which would also allow us to pinpoint the last step that was executed. In that case, we'd can explicitly state the passing outcome with the '`passed`' severity instead (and ensure that error has a higher severity as passed).